pkg/controllers/selection/controller.go

```diff
@@ -137,22 +137,22 @@ func validateTopology(pod *v1.Pod) (errs error) {
 	return errs
 }
 
-func validateAffinity(pod *v1.Pod) (errs error) {
-	if pod.Spec.Affinity == nil {
+func validateAffinity(p *v1.Pod) (errs error) {
+	if p.Spec.Affinity == nil {
 		return nil
 	}
-	if pod.Spec.Affinity.PodAffinity != nil {
+	if pod.HasPodAffinity(p) {
```

Comment:
At one point I had a PR that attached validation behavior to code modules that handled such pieces of scheduling logic. For example

topology.Validate(pod)
antiaffinity.Validate(pod)
affinity.Validate(pod)

What do you think about moving that logic instead of adding more to the `pod` library, which is a bit of a grab bag.

Reply:
Longer term I think that makes sense for when we implement support for affinity and anti-affinity. But since we don't support it currently, those modules wouldn't have anything except the validation logic. 

Can we defer on co-locating these until we implement those scheduling features? And at that time, these checks will actually be removed anyways.

Reply:
SGTM.